src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/LocalsTests.cs

```diff
@@ -3806,6 +3808,52 @@ static void Main()
             });
         }
 
+        [WorkItem(298297, "https://devdiv.visualstudio.com/DevDiv/_workitems?id=298297")]
+        [Fact]
+        public void OrderOfArguments_ArgumentsOnly()
+        {
+            var source =
+@"using System.Collections.Generic;
+class C
+{
+    static IEnumerable<object> F(object y, object x)
+    {
+        yield return x;
+#line 500
+        DummySequencePoint();
+        yield return y;
+    }
+    static void DummySequencePoint()
+    {
+    }
+}";
+            var comp = CreateCompilationWithMscorlib45(source, options: TestOptions.ReleaseDll);
+            WithRuntimeInstance(comp, runtime =>
+            {
+                EvaluationContext context;
+                context = CreateMethodContext(runtime, "C.<F>d__0.MoveNext", atLineNumber: 500);
+                string unused;
+                var locals = new ArrayBuilder<LocalAndMethod>();
+                context.CompileGetLocals(locals, argumentsOnly: true, typeName: out unused, testData: null);
+                var names = locals.Select(l => l.LocalName).ToArray();
+                // The order must confirm the order of the arguments in the method signature.
+                Assert.Equal(names, new[] { "y", "x" });
```

Comment:
`locals.Free();` here and in block below.

Reply:
Thanks!

---
In reply to: [135918580](https://github.com/dotnet/roslyn/pull/21707#discussion_r135918580) [](ancestors = 135918580)